robyn/authentication.py

```diff
@@ -64,7 +64,7 @@ def unauthorized_response(self) -> Response:
         )
 
     @abstractmethod
-    def authenticate(self, request: Request) -> Optional[Identity]:
+    def authenticate(self, request: Request) -> Union[Optional[Identity], Coroutine[Any, Any, Optional[Identity]]]:
```

Comment:
Hey @IA-PieroCV 👋 

Thanks for the PR but why do we need this type?

Reply:
@IA-PieroCV 👋 

Should we use a type alias instead? The return type is not super readable